### PR TITLE
Feat/2450 optimize render

### LINKED ISSUE #2450: Optimize app ui rendering

### PR DESCRIPTION
**Pull request by crui3er**
Closes #2450

### CODE REVIEW

File: src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                .WithTransiencyResolver(TransiencyResolvers.PreferTransient)
+                .Log(LogLevel.Debug, Log)
+                .RetryForever(retryDelays, Log)
+            ).RunIsolated(cancellationToken);
+    }
+
+    private Task TrackScreenSize(CancellationToken cancellationToken)
+        => Dispatcher.InvokeAsync(async () => {
+            var lastIsWide = IsWide();
+            await foreach (var _ in ScreenSize.Changes(cancellationToken).ConfigureAwait(false)) {
```
Comment (frolyo) [inline src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs]: why is being run in UI thread?

File: src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        });
+
+    private Task TrackRightPanelSearchMode(CancellationToken cancellationToken)
+        => Dispatcher.InvokeAsync(async () => {
+            await foreach (var isSearchMode in Right.IsSearchMode.Changes(cancellationToken).ConfigureAwait(false)) {
+                if (!isSearchMode.Value)
+                    continue;
+
+                // Open right panel when search mode is triggered on.
+                await Dispatcher
```
Comment (frolyo) [inline src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs]: what is the trick with Dispatching inside dispatching?

**Review by frolyo — APPROVED** [review]

File: src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs (hunk 2 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs]: I used the same approach for tracking screen size. I also doubted whether remove this extra dispatch or leave as it is. And last option won finally.

File: src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.StateSync.cs]: I've just move this code from [src/dotnet/UI.Blazor/Services/PanelsUI/PanelsUI.cs](https://github.com/Actual-Chat/actual-chat/pull/2470/files#diff-21601a4e79fcf355b7dd2271e31d51f2127d4fba2602d7d129bbed867bf1f270)

**Pull request merged**